src/ride/controllers/search_ride_controller.py

```diff
@@ -0,0 +1,29 @@
+from api.utils import failure_response
+from api.utils import success_response
+from api.utils import update
+from django.core.exceptions import ObjectDoesNotExist
+from person.models import Person
+
+from ..models import Ride
+from ..models import Path
+
+class SearchRideController:
+    def __init__(self, data, request, serializer): 
+        self._request = request
+        self._data = data
+        self._serializer = serializer
+
+    def process(self):
+        departure_datetime = self._data.get("departure_datetime")
+        start_location_name = self._data.get("start_location_name")
+        end_location_name = self._data.get("end_location_name")
+        if not (Path.objects.filter(start_location_name = start_location_name, end_location_name = end_location_name)).exists():
+            return failure_response("Path does not exist.")
+        
+        path = Path.objects.get(start_location_name = start_location_name, end_location_name = end_location_name)
```

Comment:
could you filter the name by using adding a __iexact to the start/end location name? this ensures we still get all results regardless of case 

https://docs.djangoproject.com/en/4.0/ref/models/querysets/#iexact

Reply:
nice